PublicAPI/Microsoft.ApplicationInsights.dll/netstandard1.3/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,15 @@
+Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.DiagnosticsTelemetryModule.AddHealthProperty(string propertyName, string propertyValue, bool isHealthy) -> bool
```

Comment:
I remember that the latest agreement was to make sure we do not ship this feature with the public API in the first version of it and make it work only for the internal modules in the first beta-versions and then start making it public based on our learnings from beta and on the feature set we would like to add over the course of the betas.
Currently, unshipped Public API lists 15 new entries to our public surface with this PR - is it intentional and the required bare minimum to make it work internally or those are the things we can still make internal?

E.g. `AddHealthProperty()` seems to be a public method of the public class in the current PR.

Reply:
Yes, we are to keep this for _module authors only_, but not authors within the -dotnet repo only. @Sergey asked me to show how someone outside our repo could extend heartbeat properties using the Web module as a good example. Also, we still need to allow customers some degree of control over the feature as it will impact their monthly quota, however minimal.

I've exposed the following properties to allow for the customer to set the fields at construction time:
`public bool IsHeartbeatEnabled`
`public IList<string> ExcludedHeartbeatProperties`
`public TimeSpan HeartbeatInterval`
...setting them back to internal makes it impossible to set via the ApplicationInsights.config file if I am not mistaken, removing any sort of control over our heartbeat from any customer. (Please correct me if I am mistaken here).

I've exposed the following methods to allow module authors (I made use of the dotnet-server repo to create my sample) to extend the properties of the heartbeat:
`public AddHealthProperty(...)`
`public SetHealthProperty(...)`

If I make these internal, then the methods are never exposed to anyone outside of the dotnet repository/project. This would make it impossible for anyone to extend the heartbeat at all, including module authors, outside of this project.

Lastly, I created the IHeartbeatPropertyManager interface to encapsulate all of this functionality so that we could rearrange things later on, and *not* depend on references directly to `DiagnosticsTelemetryModule` for those that wish to extend heartbeat properties during the beta (hopefully avoiding any major refactoring on their part once we go GA with the feature?).

Let me know if this satisfies the 'Changes in Public Surface reviewed' from the PR checklist above (I've unchecked it as of this morning since there doesn't seem to be full consensus here yet).

Reply:
This beta is the last before stable release of 2.5, so any API that makes it into this beta makes it into the stable, I would just like to be 100% certain that we stripped it to the bare minimum we require ourselves for the successful operation. Once released in stable 2.5 we'll only be able to deprecate or introduce a breaking change within this API in 3.0. As per you explanation it looks like it's the case and we are at the minimum. I remember we were discussing `InternalsVisibleTo` as the last resort to avoid any public API, but if surface area is small enough it might not be worth to overcomplicate it like this. @SergeyKanzhelev to provide any other comments on the current API.